src/components/HorizontalScroll.js

```diff
@@ -3,6 +3,22 @@ import './HorizontalScroll.scss';
 import classnames from 'classnames';
 import useVisibility from 'react-use-visibility';
 
+/**
+ * Keep track of a global timer for nudging the user to view the
+ * right scroll button. This resets every 10min starting after the
+ * first time it is animated.
+ */
+let showAnimation = true;
+let animationTimer = false;
+const didAnimate = () => {
```

Comment:
Interesting solution! @davidsmejia is there any disadvantage of having this global variable here? I can't think of any now, but seems strange.

Reply:
Ya know at the time I pushed that it seemed fine. Since I was solving for keeping the value consistent across multiple pages and ```horizontalScroll``` instances. 

You'd think I never heard of ```localstorage```  before.

How about something like this...

This will fire during a leading time of 5s and then expire and re-fire after 10min. 

Thoughts?

```javascript

function ButtonRight({ onClick, disabled }) {
  const ref = useRef();
  const isVisible = useVisibility(ref.current);
  const now = (new Date()).getTime();
  const [animationTime, setAnimaionTime] = useLocalStorage(
    `horizontalScroll/animationTime`
  );
  const wasUndefined = typeof animationTime === 'undefined';
  const lastAnimationAt = animationTime || now;
  const leading = now - lastAnimationAt <= (5 * 1000);
  const trailing = now - lastAnimationAt > (10 * 60 * 1000);
  const showAnimation = leading || trailing;
  const animate = showAnimation && (isVisible && !disabled);

  React.useEffect(() => {
    if (animate && leading && wasUndefined) {
      setAnimaionTime(now);
    }
    if (animate && trailing) {
      setAnimaionTime(now);
    }
  }, [animate, wasUndefined, leading, trailing, now, setAnimaionTime]);

  return (
    <button
      type="button"
      className={classnames({
        'horizontal-scroll__right': true,
        'horizontal-scroll__disabled': disabled,
      })}
      onClick={onClick}
    >
      <div
        ref={ref}
        className={classnames({
          'horizontal-scroll__button': true,
          'horizontal-scroll__button--animate': animate,
        })}
      >
        {'>'}
      </div>
    </button>
  );
}

```

Reply:
Actually in theory we could also change that leading time to something that would only be true if the user saw the initial animation by biding to ```animationend``` so no risk of it firing multiple times during that (in this case 5s)

Reply:
I'm not sure I understand fully how does this works... so after the component is loaded it will wait for 5secs to trigger the animation? and then wait at least another 10mins before triggering it again?

Reply:
Assuming that the button is visible and not disabled...

The animation class would be applied immediately then it would be removed after 5 seconds. If the animation class was applied 10min ago or longer it would effectively reset and apply the animation class immediately (by updating the ```animationTime``` in localStorage).

Reply:
I see, yeah I think that works! Is it possible to abstract the logic that decides whether to animate or not into a custom function/hook? Looks kind of complicated between the effect and `trailing`/`leading` variables, but that might be just me.

Reply:
Yeah thats a great idea. I think that would be a much more elegant solution. I'll merge in the above changes then abstract it out in a subsequent commit.